### PR TITLE
fix: miscellaneous fixes for shares, artwork resize, auth limits, and watcher start

### PR DESCRIPTION
**Pull request by deluan**
### Description

A batch of small, independent fixes found while auditing a few edge cases. Each one is its own commit with a test that reproduces the problem:

- **Artwork resize decodes uncapped images.** `resizeStaticImage` used a raw `image.Decode`, so a tiny file declaring huge dimensions forced a multi-gigabyte allocation on the serve-time resize path. It now goes through the same `decodeCapped` guard the processor already uses.
- **Share creation only validated the first resource ID.** The remaining IDs were never resolved, so a non-existent or hidden entity could ride along behind a valid first one, and IDs of different kinds were accepted as one share. Every ID is now resolved as the current user and must be of the same kind.
- **Album and media file shares ignored the owner's library access.** Artist and playlist shares were already loaded as the share owner, but album and media file shares used the request context, which has no user when a public share is rendered. Metadata from libraries the owner cannot see was listed (streaming was already blocked). All share types now load as the owner.
- **Public share page returned a 500 when a media file share had no visible tracks.** `Share.CoverArtID` picked a random track without checking that any were loaded, which happens when the files went missing or the owner lost access to their library. It now returns an empty artwork ID so the page renders with the placeholder.
- **Login, createAdmin and the Jellyfin login had no request body limit.** A small body-limit middleware caps the payload at 8KiB on the `/auth` route group and on the Jellyfin `AuthenticateByName` route. `createAdminUser` also returns the datastore error instead of swallowing it.
- **Log file was created world-readable.** It is now created with mode 0600. Existing files keep their mode.
- **Last.fm auth token was logged on failure.** The field is dropped; the request ID is enough to correlate.
- **Fresh databases failed when `MusicFolder` contained a single quote.** The library migration interpolated the path into SQL with `fmt.Sprintf`; it now binds it as a parameter.
- **`getScanStatus` reported `count=0` for scans not started through the API.** Every wire injector built its own scanner controller, and `Status` reads the progress counters from its own instance. Startup, periodic, watcher and SIGUSR1 scans ran on a different instance from the API routers. The controller is now a singleton, like the watcher and broker. Verified live with a startup scan: master reports count 0 while scanning, this branch reports the real counts.
- **Folder watcher hung when it could not start.** When `notify.Watch` failed, `Start` blocked until context cancellation and left the `watching` flag set. `notify.Watch` now runs before the event loop goroutine, so the error is returned immediately.

### Type of Change

- [x] Bug fix
- [ ] New feature
- [ ] Documentation update
- [ ] Refactor
- [ ] Other (please describe):

### Checklist

- [x] My code follows the project's coding style
- [x] I have tested the changes locally
- [ ] I have added or updated documentation as needed
- [x] I have added tests that prove my fix/feature works (or explain why not)
- [x] All existing and new tests pass

### How to Test

Each fix has a unit test; `make test` covers all of them. To check a few by hand:

1. **Share scoping:** as a non-admin user with access to one library, create a share for an album. Open the public share link in a private window. Only tracks from that library appear.
2. **Login body limit:** `curl -X POST localhost:4533/auth/login -d "$(python3 -c 'print("{\"username\":\"x\",\"password\":\"" + "a"*10000 + "\"}")')"` returns 422.
3. **Quoted music folder:** start with an empty data folder and `ND_MUSICFOLDER="/tmp/Rock 'n' Roll"`. The server starts and the library row holds the path unchanged.
4. **Log file mode:** start with `ND_LOGFILE=/tmp/nd.log` and check `ls -l /tmp/nd.log` shows `-rw-------`.

### CODE REVIEW

**Comment by deluan** [comment]
@codex review

**Comment by deluan** [comment]
@codex review

**Comment by deluan** [comment]
@codex review

File: server/auth.go [hunk 1]
```diff
@@ -96,6 +96,16 @@ func buildAuthPayload(user *model.User) map[string]any {
 	return payload
 }
 
+// MaxLoginBodySize bounds the payload of unauthenticated login routes across all APIs.
+const MaxLoginBodySize = 8 << 10
+
+func LimitLoginBody(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		r.Body = http.MaxBytesReader(w, r.Body, MaxLoginBodySize)
+		next.ServeHTTP(w, r)
```
[the remark this answers is not in the document]
Comment (deluan) [reply server/auth.go:105]: I checked this one. The cap exists to bound what the server spends decoding an unauthenticated body, and it does that: with a valid credentials object followed by 8 KiB, 1 MiB or 64 MiB of trailing bytes, the handler reads 512 bytes from the socket in every case, because json.Decoder stops reading once the first value is complete. The trailing bytes are never allocated. After the handler returns, net/http either discards up to 256 KiB or closes the connection without reading, so the leftover costs nothing on our side. The two suggested fixes would make it worse: requiring EOF moves a blocking read into the handler and gives a slow client a way to hold the login goroutine, and a Content-Length check is bypassed by chunked encoding. Keeping it as is.

File: persistence/share_repository.go [hunk 2]
```diff
@@ -127,7 +121,7 @@ func (r *shareRepository) loadMedia(share *model.Share) error {
 		share.Tracks = tracks.MediaFiles()
 		return nil
 	case "media_file":
-		mfRepo := NewMediaFileRepository(r.ctx, r.db)
+		mfRepo := NewMediaFileRepository(ctx, r.db)
 		tracks, err := mfRepo.GetAll(model.QueryOptions{Filters: noMissing(Eq{"media_file.id": ids})})
 		share.Tracks = sortByIdPosition(tracks, ids)
```
[the remark this answers is not in the document]
Comment (deluan) [reply persistence/share_repository.go:126]: Good catch on the panic, though it is older than this PR: on master the media file branch already filters missing tracks, and share.resource_ids has no foreign key, so a share whose files were removed hits the same random pick on an empty slice. Owner scoping adds one more way to get there. Fixed at the source in 82bcf02f68: CoverArtID now returns an empty artwork ID when there are no tracks, so every caller gets the placeholder instead of a 500, and the dead guard on the split IDs is gone.

**Comment by deluan** [comment]
@codex review

**Comment by deluan** [comment]
@codex review
